Transfer/Tests/CreateTransferPresenterTests.swift

```diff
@@ -9,73 +9,74 @@ import XCTest
 class CreateTransferTests: XCTestCase {
     private var presenter: CreateTransferPresenter!
     private var mockView = MockCreateTransferView()
-
+    
     private let clientTransferId = "6712348070812"
     private let sourceToken = "trm-123456789"
-
+    
     override func setUp() {
         Hyperwallet.setup(HyperwalletTestHelper.authenticationProvider)
     }
-
+    
     override func tearDown() {
         if Hippolyte.shared.isStarted {
             Hippolyte.shared.stop()
         }
         mockView.resetStates()
     }
-
-    private func initializePresenter(emptyTransferMethods: Bool) {
-        if emptyTransferMethods {
-            TransferMethodRepositoryRequestHelper.setupNoContentRequest()
-        } else {
-            TransferMethodRepositoryRequestHelper.setupSucessRequest()
+    
+    private enum LoadTransferMethodsResultType {
+        case success, failure, noContent
+        func setUpRequest() {
+            switch self {
+            case .success:
+                TransferMethodRepositoryRequestHelper.setupSucessRequest()
+                
+            case .failure:
+                TransferMethodRepositoryRequestHelper.setupFailureRequest()
+                
+            case .noContent:
+                TransferMethodRepositoryRequestHelper.setupNoContentRequest()
+            }
         }
+    }
+    
+    private func initializePresenter(transferMethodResult: LoadTransferMethodsResultType) {
+        transferMethodResult.setUpRequest()
         UserRepositoryRequestHelper.setupSucessRequest()
         CreateTransferRequestHelper.setupSucessRequest()
-
+        
         presenter = CreateTransferPresenter(clientTransferId, nil, view: mockView)
-
+        
         let loadCreateTransferExpectation = self.expectation(description: "loadCreateTransferExpectation")
         mockView.expectation = loadCreateTransferExpectation
-
+        
         presenter.loadCreateTransfer()
         wait(for: [loadCreateTransferExpectation], timeout: 1)
         presenter.initializeSections()
     }
-
+    
     private func initializePresenterWithSource() {
         presenter = CreateTransferPresenter(clientTransferId, sourceToken, view: mockView)
         presenter.loadCreateTransfer()
     }
-
+    
     private func initializePresenterEmptyTransferMethods() {
         let response = HyperwalletTestHelper.noContentHTTPResponse()
         let url = String(format: "%@%@", HyperwalletTestHelper.userRestURL, "/transfer-methods?")
         HyperwalletTestHelper.setUpMockServer(request:
             HyperwalletTestHelper.buildGetRequestRegexMatcher(pattern: url, response))
     }
-
+    
     func testSectionDataWhenSelectedTransferMethodIsNil() {
-        initializePresenter(emptyTransferMethods: true)
```

Comment:
don't override this test

Reply:
only `initializePresenter(transferMethodResult: .noContent)`

Reply:
Sorry, I don't understand "**don't override this test"**

Reply:
@ymoskovko-epam , I got it.